fendermint/vm/topdown/src/sync/syncer.rs

```diff
@@ -119,14 +91,27 @@ where
                 break;
             }
 
-            let synced_height = self.poll_next().await?;
+            let synced_height = match self
+                .poll_next(latest_height_fetched + 1, first_non_null_parent_hash)
```

Comment:
Maybe we can massage the return value so that instead of returning the height, which we don't need now, since we know that it's going to be `latest_height_fetched + 1`, we return the _next parent hash_ to use, which will the the same as the input if we found a null round, or it's going to be the hash of the non-null block we fetched. With this value you can eliminate the call to `get_from_cache()` at the end of the loop.